runner-starter/src/main/java/org/apache/apisix/plugin/runner/PluginRunnerApplication.java

```diff
@@ -17,17 +17,138 @@
+                        allFilters[i] = allFilters[i].substring(0, allFilters[i].length() - 5);
+                        set.add(allFilters[i]);
+                    }
+                }
+
+                for (String filterName : allFilters) {
+                    if ((!filterName.equals("package-info.java")) && filterName.charAt(filterName.length() - 1) != '~') {
+                        //Bean Filter Name necessary because beans always start with lower case letters
+                        String beanFilterName = Character.toLowerCase(filterName.charAt(0)) + filterName.substring(1);
+                        if (registry.containsBeanDefinition(beanFilterName)) {
+                            registry.removeBeanDefinition(beanFilterName);
+                        }
+                        JavaCompiler compiler = ToolProvider.getSystemJavaCompiler();
+                        String[] args = {"-d", pathToProject + "/target/classes", absolutePath + filterName + ".java"};
+                        compiler.run(null, null, null, args);
+
+                        CLASS_LOADER = new DynamicClassLoader(PARENT_CLASS_LOADER);
+                        CLASS_LOADER.setDir(pathToProject + "/target/classes");
+                        CLASS_LOADER.setFilters(set);
+                        CLASS_LOADER.setPackageName(packageName);
+                        Class myObjectClass = CLASS_LOADER.loadClass(filterName);
+                        BeanDefinitionBuilder builder = BeanDefinitionBuilder.genericBeanDefinition(myObjectClass).setLazyInit(true);
+                        registry.registerBeanDefinition(beanFilterName, builder.getBeanDefinition());
+                    }
+                }
+
+                //removes a filter dynamically
+                List<String> allRemovedFilters = findRemovedFilters(pathToProject, packageName, set);
+                for (String removedFilter : allRemovedFilters) {
+                    String beanRemovedFilter = Character.toLowerCase(removedFilter.charAt(0)) + removedFilter.substring(1);
+                    if (registry.containsBeanDefinition(beanRemovedFilter)) {
+                        registry.removeBeanDefinition(beanRemovedFilter);
+                    }
+                }
+
+            }
+            poll = key.reset();
+        }
+    }
+
+    public List<String> findRemovedFilters(String pathToProject, String packageName, HashSet<String> set) {
+        List<String> allRemovedFilters = new ArrayList<>();
+        String packagePath = packageName.replaceAll("\\.", "/");
+        String[] allClasses = new File(pathToProject + "/target/classes/" + packagePath + "/").list();
+        for (String className : allClasses) {
+            className = className.substring(0, className.length() - 6);
+            if (!set.contains(className)) {
+                allRemovedFilters.add(className);
+            }
+        }
+        return allRemovedFilters;
+    }
```

Comment:
Is there any way we can put the added content in a separate class?

Reply:
Can you clarify what you mean by this? Not sure what you meant by added content.

Reply:
done, put new functionality in class called HotReloadProcess